lib/oban/testing.ex

```diff
@@ -214,6 +222,11 @@ defmodule Oban.Testing do
     |> apply_where_clauses(fields_with_opts)
   end
 
+  defp extract_prefix(opts) do
+    {prefix, opts} = Keyword.pop(opts, :prefix)
```

Comment:
No need for the `||` below, you can provide a default to `pop/3`:

```suggestion
    {prefix, opts} = Keyword.pop(opts, :prefix, "public")
```